### ISSUE TITLE
"wait for element present" ignores specified timeout value

### ISSUE THREAD

**Issue by swartzmc** [SeleniumHQ/selenium-ide#1466]
## 🐛 Bug Report

Using the Selenium IDE 3.17.4 Firefox extension with 64-bit Firefox 91.10.0ESR on 64-bit Windows 10 21H2 (OS BUild 19044.1706) on a very slow website...

The timeout value specified in the "wait for element present" command is ignored.

Given the following command:

wait for element present  id=seeAllOfferLink 180000

Selenium IDE should wait up to 3 minutes for the element with ID=seeAllOfferLink to display, but instead fails after the default 30 seconds have elapsed. 

I can't provide the full SIDE file for confidentiality reasons, but the relevant portion is:

{
      "id": "c6cb5979-534d-4bbc-9fc3-72df398dfc9c",
      "comment": "",
      "command": "waitForElementPresent",
      "target": "id=seeAllOfferLink",
      "targets": [
        ["id=seeAllOfferLink", "id"],
        ["linkText=See All Offers", "linkText"],
        ["css=.textCenter:nth-child(2) > #seeAllOfferLink", "css:finder"],
        ["xpath=//a[contains(text(),'See All Offers')]", "xpath:link"],
        ["xpath=//a[@id='seeAllOfferLink']", "xpath:attributes"],
        ["xpath=//div[@id='purchasePaybackBottomRow']/div[2]/div/a", "xpath:idRelative"],
        ["xpath=(//a[contains(@href, '#')])[22]", "xpath:href"],
        ["xpath=//div[2]/div[2]/div/a", "xpath:position"],
        ["xpath=//a[contains(.,'See All Offers')]", "xpath:innerText"]
      ],
      "value": "180000"
    }

The delay in displaying this element is due to the host website spending several minutes trying to contact tracking/advertising sites in the background that are being blocked by a firewall. The first half of the page displays immediately, but the portion containing the button with ID=seeAllOfferLink can take up to 3 minutes to load before the web page stops retrying those sites. The wait command fails before the page finishes loading. I am currently using "pause" as a workaround.

**Comment by toddtarsi** [comment]
We're out in Alpha on v4! Please try there and let me know what you see. If it's still broken, open a new issue!

https://github.com/SeleniumHQ/selenium-ide/releases/tag/4.0.0-alpha.0
